### ISSUE TITLE
Does not support CTRL-U in password mode

### ISSUE THREAD

**Issue by gutenye** [JEG2/highline#43]
Type "helloCTRL-Uworld"

``` ruby
ask("Usrename: ")                            -> "world"
ask("Password: "){|q| q.echo = false}  -> "hello\x15world"
```

**Comment by abinoam** [comment]
Hi @GutenYe , just to be sure I'm understanding correctly. Are you expecting HighLine to "delete" all the line and come back to "home", so it would return only "world", not "hello\u0015world" ?

If so, I think we could try to do this. 
But, should we try to emulate all control keys from the terminal?

For most common purposes of the **echo = false** use cases, handling backspace correctly is enough.

**Comment by JEG2** [comment]
Yeah, I'm not sure how much it's HighLine's responsibility to emulate full terminal behavior.  It seems weird to me that a simple HighLine call might erase my terminal screen.

We are currently planning the next major version of HighLine.  We will include a discussion of where to draw these lines in the plans we make for that release.

**Comment by GutenYe** [comment]
> Are you expecting HighLine to "delete" all the line 

Yes, that's what I wanted.

**Comment by abinoam** [comment]
Hi @gutenye,

Most projects that I know just delete (deprecate) old issues. But it's not the case here in HighLine. We don't give up! :smile: 
This issue has **more than 10 years**! Oh - MY - GOD!

The first release of "io/console" was 0.3 at October 16, 2011 (6,5 KB). But, not mature, and in a time that some of the releases were yanked.

The first "usable" release that I have notice is 0.4.1 at February 04, 2013 (8,5 KB) **AFTER** the opening of this issue.

You can see the whole release history at https://rubygems.org/gems/io-console/versions

The good news is that we currently have an "io/console" more mature with more features we could rely on so we don't have to take all the responsibilities for the console inside HighLine's code.

I've just merged a PR that solves handling "CTRL-C" by relying on "io/console" functionalities that were released with Ruby version 2.7.0 at December, 2019 (when this issue had 7 years old.
Then I remembered about this old issue over here and this improvements in "io/console" gave me some hope.
I can't promise, but I'll be trying to solve it.

Is there any GitHub badge for solving a 10 years old issue? :smile: 

Hey @JEG2 keep an eye on this? Perhaps we take a picture of an "anniversary cake" for the issue.

**Comment by abinoam** [comment]
It was easier to treat CTRL-U as we are treating backspace now than to rely on `io/console`. Basically because of things like the possibility of `echo = '*'` for example, or `question.limit`. But we surely have to transition on to relying on `io/console` for everything it is possible now. Se PR #260
